app/js/trouble.js

```diff
@@ -56,12 +58,16 @@
 				for (i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+      };
+			var timeId= setTimeout(function() { setDropdownHeight(); }, 1000);

```

Comment:
This does get the "making" of a function outside the loop, but there's a couple of other things to keep in mind here. The return value of setTimeout is just a number, so it can't be invoked. The best solution here is to either pass a reference to the function as the first argument to setTimeout (either the function name if you simply declare it or the variable name if you assign the anonymous function expression to a variable) or to just remove the infinite loop entirely.